src/borg/cache.py

```diff
@@ -514,6 +514,15 @@ def __init__(
         self.txn_active = False
 
         self.path = cache_dir(self.repository, path)
+        if os.path.exists(self.path) and not self.check_cache_integrity():
```

Comment:
isn't the first term redundant as the check call would also return False in that case?